pkg/services/login/loginservice/loginservice.go

```diff
@@ -310,12 +310,14 @@ func (ls *Implementation) syncOrgRoles(ctx context.Context, usr *user.User, extU
 				logger.Error(err.Error(), "userId", cmd.UserID, "orgId", cmd.OrgID)
 				continue
 			}
-			if err := ls.accessControl.DeleteUserPermissions(ctx, orgId, cmd.UserID); err != nil {
-				logger.Warn("failed to delete permissions for user", "userID", cmd.UserID, "orgID", orgId)
-			}
 
 			return err
 		}
+
+		if err := ls.accessControl.DeleteUserPermissions(ctx, orgId, cmd.UserID); err != nil {
+			logger.Error("failed to delete permissions for user", "error", err, "userID", cmd.UserID, "orgID", orgId)
+			return err
```

Comment:
Do you think it correct to return and stop the authentication flow if the Delete is unsuccessful ?

Reply:
Not sure about that, I seems to be what is happening if we fail to remove user from org as well

Reply:
Hmm, I feel like it might not make sense, as in the next call the user will get authenticated anyway without these permissions being removed (as this org won't be in the org list anymore because that part of deletion succeeded).

I think we'll need to introduce some sort of permission cleanup process at some point to handle cases like this.

Reply:
:+1: I'll make it back to a warning, thanks for the feedback